Linking to libopmcommon fails

Since the merge of #1686, OpenMP apparently is a hard dependency of opm-common. However, `-fopenmp` is not added to the link options automatically which makes my Dumux application fail with
```
cd /home/bernd/Dumux/dumux/build-gcc/test/porousmediumflow/2p/implicit/cornerpoint && /usr/bin/cmake -E cmake_link_script CMakeFiles/test_2p_cornerpoint.dir/link.txt --verbose=1
/usr/bin/c++  -std=c++17  -fext-numeric-literals  -fdiagnostics-color=always -fno-strict-aliasing -fstrict-overflow -fno-finite-math-only -DNDEBUG=1 -O3 -march=native -funroll-loops -g0 -Wall -Wunused -Wmissing-include-dirs -Wcast-align -Wno-missing-braces -Wmissing-field-initializers -Wno-sign-compare   -Wl,-no-as-needed -pthread  -Wl,-no-as-needed -pthread -rdynamic CMakeFiles/test_2p_cornerpoint.dir/main.cc.o  -o test_2p_cornerpoint -Wl,-rpath,/usr/lib/x86_64-linux-gnu/openmpi/lib /home/bernd/Dumux/dune-alugrid/build-gcc/lib/libdunealugrid.a /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so -pthread -lparmetis -ltrilinos_zoltan -lboost_unit_test_framework -lboost_system /home/bernd/Dumux/opm-common/build-gcc/lib/libopmcommon.a /home/bernd/Dumux/dune-grid/build-gcc/lib/libdunegrid.a /home/bernd/Dumux/dune-common/build-gcc/lib/libdunecommon.a /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_cxx.so -lboost_date_time -lmetis -lparmetis -lrt -lsuitesparseconfig -lm -lopenblas -lamd -lumfpack /home/bernd/Dumux/dune-uggrid/build-gcc/lib/libugS3.a /home/bernd/Dumux/dune-uggrid/build-gcc/lib/libugS2.a /home/bernd/Dumux/dune-uggrid/build-gcc/lib/libugL.a /home/bernd/Dumux/dune-geometry/build-gcc/lib/libdunegeometry.a -lpthread /home/bernd/Dumux/opm-grid/build-gcc/lib/libopmgrid.a -lboost_unit_test_framework -lboost_system /home/bernd/Dumux/opm-common/build-gcc/lib/libopmcommon.a /home/bernd/Dumux/dune-grid/build-gcc/lib/libdunegrid.a /home/bernd/Dumux/dune-uggrid/build-gcc/lib/libugS3.a /home/bernd/Dumux/dune-uggrid/build-gcc/lib/libugS2.a /home/bernd/Dumux/dune-uggrid/build-gcc/lib/libugL.a /home/bernd/Dumux/dune-geometry/build-gcc/lib/libdunegeometry.a /home/bernd/Dumux/dune-common/build-gcc/lib/libdunecommon.a -pthread /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so -lmetis -lm -ltrilinos_zoltan -lparmetis -pthread /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so -lparmetis -pthread /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so -pthread /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so -lz -lldl -lspqr -lumfpack -lcholmod -lamd -lcamd -lcolamd -lccolamd -lsuitesparseconfig -larpack++ -larpack -lsuperlu -lopenblas -lparmetis -pthread /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so -lmetis -lm -lmetis -lm /home/bernd/Dumux/dune-grid/build-gcc/lib/libdunegrid.a /home/bernd/Dumux/dune-uggrid/build-gcc/lib/libugS3.a /home/bernd/Dumux/dune-uggrid/build-gcc/lib/libugS2.a /home/bernd/Dumux/dune-uggrid/build-gcc/lib/libugL.a /home/bernd/Dumux/dune-geometry/build-gcc/lib/libdunegeometry.a /home/bernd/Dumux/dune-common/build-gcc/lib/libdunecommon.a -lparmetis -pthread /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so -lmetis -lm -lmetis -lm -Wl,-Bstatic -lVc -Wl,-Bdynamic -ltbbmalloc -ltbb -lquadmath -lgmp -lgmpxx -lopenblas -lopenblas -pthread /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so /home/bernd/Dumux/dune-alugrid/build-gcc/lib/libdunealugrid.a /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so -pthread -lparmetis -ltrilinos_zoltan -lboost_unit_test_framework -lboost_system /home/bernd/Dumux/opm-common/build-gcc/lib/libopmcommon.a /home/bernd/Dumux/dune-grid/build-gcc/lib/libdunegrid.a /home/bernd/Dumux/dune-common/build-gcc/lib/libdunecommon.a /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_cxx.so -lboost_date_time -lmetis -lparmetis -lrt -lsuitesparseconfig -lm -lopenblas -lamd -lumfpack /home/bernd/Dumux/dune-uggrid/build-gcc/lib/libugS3.a /home/bernd/Dumux/dune-uggrid/build-gcc/lib/libugS2.a /home/bernd/Dumux/dune-uggrid/build-gcc/lib/libugL.a /home/bernd/Dumux/dune-geometry/build-gcc/lib/libdunegeometry.a -lpthread /home/bernd/Dumux/opm-grid/build-gcc/lib/libopmgrid.a -lboost_unit_test_framework -lboost_system /home/bernd/Dumux/opm-common/build-gcc/lib/libopmcommon.a /home/bernd/Dumux/dune-grid/build-gcc/lib/libdunegrid.a /home/bernd/Dumux/dune-uggrid/build-gcc/lib/libugS3.a /home/bernd/Dumux/dune-uggrid/build-gcc/lib/libugS2.a /home/bernd/Dumux/dune-uggrid/build-gcc/lib/libugL.a /home/bernd/Dumux/dune-geometry/build-gcc/lib/libdunegeometry.a /home/bernd/Dumux/dune-common/build-gcc/lib/libdunecommon.a -pthread /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so -lmetis -lm -ltrilinos_zoltan -lparmetis -pthread /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so -lparmetis -pthread /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so -pthread /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so -lz -lldl -lspqr -lumfpack -lcholmod -lamd -lcamd -lcolamd -lccolamd -lsuitesparseconfig -larpack++ -larpack -lsuperlu -lopenblas -lparmetis -pthread /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so -lmetis -lm -lmetis -lm /home/bernd/Dumux/dune-grid/build-gcc/lib/libdunegrid.a /home/bernd/Dumux/dune-uggrid/build-gcc/lib/libugS3.a /home/bernd/Dumux/dune-uggrid/build-gcc/lib/libugS2.a /home/bernd/Dumux/dune-uggrid/build-gcc/lib/libugL.a /home/bernd/Dumux/dune-geometry/build-gcc/lib/libdunegeometry.a /home/bernd/Dumux/dune-common/build-gcc/lib/libdunecommon.a -lparmetis -pthread /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so -lmetis -lm -lmetis -lm -Wl,-Bstatic -lVc -Wl,-Bdynamic -ltbbmalloc -ltbb -lquadmath -lgmp -lgmpxx -lopenblas -lopenblas -pthread /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so -lmetis -lm -lz -ltbbmalloc -ltbb 
/usr/bin/ld: /home/bernd/Dumux/opm-common/build-gcc/lib/libopmcommon.a(EclipseGrid.cpp.o): undefined reference to symbol 'omp_get_thread_num@@OMP_1.0'
/usr/bin/ld: /lib/x86_64-linux-gnu/libgomp.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```

While I can solve this downstream by adding
```
target_link_options(test_2p_cornerpoint PUBLIC -fopenmp)
```
a fix in `opm-common` seems more appropriate.

Unfortunately, this gives me
```
CMake Error at /home/bernd/Dumux/dune-common/cmake/modules/DuneTestMacros.cmake:336 (add_executable):
  Target "test_2p_cornerpoint" links to target "OpenMP::OpenMP_CXX" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?
Call Stack (most recent call first):
  cmake/modules/DumuxTestMacros.cmake:245 (dune_add_test)
  test/porousmediumflow/2p/implicit/cornerpoint/CMakeLists.txt:4 (dumux_add_test)
```

--- a/cmake/Modules/UseOpenMP.cmake
+++ b/cmake/Modules/UseOpenMP.cmake
@@ -42,7 +42,13 @@ macro (find_openmp opm)
   # enabling OpenMP is supposedly enough to make the compiler link with
   # the appropriate libraries
   find_package (OpenMP ${${opm}_QUIET})
-  list (APPEND ${opm}_LIBRARIES ${OpenMP_LIBRARIES})
+
+  if(OpenMP_CXX_FOUND)
+    list (APPEND ${opm}_LIBRARIES OpenMP::OpenMP_CXX)
+  else()
+    list (APPEND ${opm}_LIBRARIES ${OpenMP_LIBRARIES})
+  endif()
+
   if (OPENMP_FOUND)
 	add_options (C ALL_BUILDS "${OpenMP_C_FLAGS}")
 	add_options (CXX ALL_BUILDS "${OpenMP_CXX_FLAGS}")

